### PR TITLE
KAFKA-17060: Rename LegacyKafkaConsumer to ClassicKafkaConsumer

### PR DESCRIPTION
**Pull request by kirktrue**
Rename `LegacyKafkaConsumer` to `ClassicKafkaConsumer` to avoid confusion.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/ClassicKafkaConsumer.java [hunk 1]
```diff
@@ -146,7 +146,7 @@ public class LegacyKafkaConsumer<K, V> implements ConsumerDelegate<K, V> {
     // to keep from repeatedly scanning subscriptions in poll(), cache the result during metadata updates
     private boolean cachedSubscriptionHasAllFetchPositions;
 
-    LegacyKafkaConsumer(ConsumerConfig config, Deserializer<K> keyDeserializer, Deserializer<V> valueDeserializer) {
+    ClassicKafkaConsumer(ConsumerConfig config, Deserializer<K> keyDeserializer, Deserializer<V> valueDeserializer) {
```
Comment (lianetm) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/ClassicKafkaConsumer.java:153]: we should also remove the reference to LegacyKafkaConsumer right above, on ln 140

Comment (kirktrue) [reply clients/src/main/java/org/apache/kafka/clients/consumer/internals/ClassicKafkaConsumer.java:153]: Good call! I'll look for more text matches.

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/ClassicKafkaConsumer.java [hunk 2]
```diff
@@ -107,7 +107,7 @@
  * group protocol (from KIP-848) introduces allows users continue using the legacy "classic" group protocol.
```
Comment (lianetm) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/ClassicKafkaConsumer.java]: nit: introduces allow...not introduced here but makes sense to review the javadoc after the rename

**Comment by lianetm** [comment]
Hey @kirktrue , thanks for the patch, nice! Just minor comments.

**Comment by chia7712** [comment]
The failure will be fixed by #16688

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by chia7712** [comment]
@kirktrue Could you please merge trunk to include fix #16688? I feel this patch is good but it would be nice to see CI again :)

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/ClassicKafkaConsumer.java [hunk 3]
```diff
@@ -103,11 +103,15 @@
  *
  * <p/>
  *
- * <em>Note:</em> per its name, this implementation is left for backward compatibility purposes. The updated consumer
- * group protocol (from KIP-848) introduces allows users continue using the legacy "classic" group protocol.
- * This class should not be invoked directly; users should instead create a {@link KafkaConsumer} as before.
+ * This {@link ConsumerDelegate} implementation exists for backward compatibility to allow users to continue to use
+ * the existing classic group protocol.
```
Comment (AndrewJSchofield) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/ClassicKafkaConsumer.java]: I think "existing" is redundant here. I suggest "classic group protocol (pre-KIP 848)".

**Review by AndrewJSchofield — CHANGES_REQUESTED** [review]
This looks like a worthwhile change to me. I'm surprised that the terminology hadn't spread more broadly, but I think you've caught all of the instances I could find.

**Review by AndrewJSchofield — APPROVED** [review]
lgtm

**Review by lianetm — APPROVED** [review]
LGTM. Let's just make sure to merge trunk latest changes as @chia7712 suggested [above](https://github.com/apache/kafka/pull/16683#issuecomment-2250035455) (last build did have those failures). Thanks @kirktrue !

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/ClassicKafkaConsumer.java (hunk 1 above)
Comment (kirktrue) [reply clients/src/main/java/org/apache/kafka/clients/consumer/internals/ClassicKafkaConsumer.java:153]: I _believe_ I've found all of the occurrences of `LegacyKafkaConsumer` in code, comments, etc.

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/ClassicKafkaConsumer.java (hunk 2 above)
Comment (kirktrue) [reply clients/src/main/java/org/apache/kafka/clients/consumer/internals/ClassicKafkaConsumer.java]: Updated. Thanks!

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/ClassicKafkaConsumer.java (hunk 3 above)
Comment (kirktrue) [reply clients/src/main/java/org/apache/kafka/clients/consumer/internals/ClassicKafkaConsumer.java]: Fixed, as suggested. Thanks!

**Pull request merged**